src/libostree/ostree-repo-pull.c

```diff
@@ -1829,6 +2002,41 @@ scan_commit_object (OtPullData                 *pull_data,
     }
 #endif /* OSTREE_DISABLE_GPGME */
 
+  if (pull_data->sign_verify &&
+      !g_hash_table_contains (pull_data->verified_commits, checksum))
+    {
+      gboolean ret = FALSE;
+      /* list all signature types in detached metadata and check if signed by any? */
+      g_auto (GStrv) names = ostree_sign_list_names();
+      for (guint i=0; i < g_strv_length (names); i++)
+        {
+          g_autoptr (OstreeSign) sign = NULL;
+          g_autoptr (GError) local_error = NULL;
+
+          if ((sign = ostree_sign_get_by_name (names[i], &local_error)) == NULL)
+            continue;
+
+          /* Try to load public key(s) according remote's configuration */
+          if (!_load_public_keys (sign, pull_data->repo, pull_data->remote_name))
+            continue;
+
+          /* Set return to true if any sign fit */
+          if (ostree_sign_commit_verify (sign,
+                                         pull_data->repo,
+                                         checksum,
+                                         cancellable,
+                                         &local_error))
+            ret = TRUE;
+
+        }
+      if (!ret)
+        {
+          g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
```

Comment:
This is ignoring `local_error`; the message there would likely be very useful right?  We could gather all errors into an array and stringify all of them, or probably simplest just retain the last one.

Reply:
Reworked error handling -- now the real reason(s) from signature engines are collected instead of just set the common error.